components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -137,6 +139,30 @@ export class PrebuildManager {
         }
     }
 
+    async cancelPrebuild(ctx: TraceContext, user: User, prebuildId: string): Promise<void> {
+        const span = TraceContext.startSpan("cancelPrebuild", ctx);
+        span.setTag("prebuildId", prebuildId);
+        try {
+            const prebuild = await this.workspaceDB.trace({ span }).findPrebuildByID(prebuildId);
+            if (!prebuild || prebuild.state === 'aborted') {
+                return;
+            }
+            // TODO(janx): stop `prebuild.buildWorkspaceId`?
```

Comment:
without stopping the actual prebuild, you'll get status updates that reset the changes below.

Reply:
Many thanks @csweichel! I was wondering whether the headless update call below would take care of that or not. Let's also wait on stopping the prebuild workspace then :+1:

Reply:
Fixed